In [1]:
%load_ext nb_black

<IPython.core.display.Javascript object>

In [2]:
CLASSIFICATIONS = {
    "01": "Agriculture,Forestry and Fishing",
    "02": "Agriculture, Forestry and Fishing",
    "03": "Agriculture, Forestry and Fishing",
    "05": "Mining and Quarrying",
    "06": "Mining and Quarrying",
    "07": "Mining and Quarrying",
    "08": "Mining and Quarrying",
    "09": "Mining and Quarrying",
    "10": "Manufacturing",
    "11": "Manufacturing",
    "12": "Manufacturing",
    "13": "Manufacturing",
    "14": "Manufacturing",
    "15": "Manufacturing",
    "16": "Manufacturing",
    "17": "Manufacturing",
    "18": "Manufacturing",
    "19": "Manufacturing",
    "20": "Manufacturing",
    "21": "Manufacturing",
    "22": "Manufacturing",
    "23": "Manufacturing",
    "24": "Manufacturing",
    "25": "Manufacturing",
    "26": "Manufacturing",
    "27": "Manufacturing",
    "28": "Manufacturing",
    "29": "Manufacturing",
    "30": "Manufacturing",
    "31": "Manufacturing",
    "32": "Manufacturing",
    "33": "Manufacturing",
    "35": "Electricity, Gand, Steam and Air Conditioning Supply",
    "36": "Water Supply; Sewerage, Waste Management and Remediation Activities",
    "37": "Water Supply; Sewerage, Waste Management and Remediation Activities",
    "38": "Water Supply; Sewerage, Waste Management and Remediation Activities",
    "39": "Water Supply; Sewerage, Waste Management and Remediation Activities",
    "41": "Construction",
    "42": "Construction",
    "43": "Construction",
    "45": "Wholesale and Retail Trade; Repair of Motor Vehicles and Motorcycles",
    "46": "Wholesale and Retail Trade; Repair of Motor Vehicles and Motorcycles",
    "47": "Wholesale and Retail Trade; Repair of Motor Vehicles and Motorcycles",
    "49": "Transportation and Storage",
    "50": "Transportation and Storage",
    "51": "Transportation and Storage",
    "52": "Transportation and Storage",
    "53": "Transportation and Storage",
    "55": "Accommodation and Food Service Activities",
    "56": "Accommodation and Food Service Activities",
    "58": "Information and Communication",
    "59": "Information and Communication",
    "60": "Information and Communication",
    "61": "Information and Communication",
    "62": "Information and Communication",
    "63": "Information and Communication",
    "64": "Financial and Insurance Activities",
    "65": "Financial and Insurance Activities",
    "66": "Financial and Insurance Activities",
    "68": "Real Estate Activities",
    "69": "Professional, Scientific and Technical Activities",
    "70": "Professional, Scientific and Technical Activities",
    "71": "Professional, Scientific and Technical Activities",
    "72": "Professional, Scientific and Technical Activities",
    "73": "Professional, Scientific and Technical Activities",
    "74": "Professional, Scientific and Technical Activities",
    "75": "Professional, Scientific and Technical Activities",
    "77": "Administrative and Support Service Activities",
    "78": "Administrative and Support Service Activities",
    "79": "Administrative and Support Service Activities",
    "80": "Administrative and Support Service Activities",
    "81": "Administrative and Support Service Activities",
    "82": "Administrative and Support Service Activities",
    "84": "Public Administration and Defence; Compulsory Social Security",
    "85": "Education",
    "86": "Human Health and Social Work Activities",
    "87": "Human Health and Social Work Activities",
    "88": "Human Health and Social Work Activities",
    "90": "Arts, Entertainment and Recreation",
    "91": "Arts, Entertainment and Recreation",
    "92": "Arts, Entertainment and Recreation",
    "93": "Arts, Entertainment and Recreation",
    "94": "Other Service Activities",
    "95": "Other Service Activities",
    "96": "Other Service Activities",
    "97": "Activities of Households and Employers",
    "98": "Activities of Households and Employers",
    "99": "Activities of Extraterritorial Organisations and Bodies",
}

<IPython.core.display.Javascript object>

In [3]:
import pandas as pd

<IPython.core.display.Javascript object>

In [4]:
df = pd.read_csv("../CH/BasicCompanyDataAsOneFile-2022-05-01.csv", low_memory=True)

C:\Users\VRUser2\AppData\Local\Temp\ipykernel_14716\1159366089.py:1: DtypeWarning: Columns (41,42,43,44,45,46,47,48,49,50,51,52) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../CH/BasicCompanyDataAsOneFile-2022-05-01.csv", low_memory=True)


<IPython.core.display.Javascript object>

In [5]:
df.columns

Index(['CompanyName', ' CompanyNumber', 'RegAddress.CareOf',
       'RegAddress.POBox', 'RegAddress.AddressLine1',
       ' RegAddress.AddressLine2', 'RegAddress.PostTown', 'RegAddress.County',
       'RegAddress.Country', 'RegAddress.PostCode', 'CompanyCategory',
       'CompanyStatus', 'CountryOfOrigin', 'DissolutionDate',
       'IncorporationDate', 'Accounts.AccountRefDay',
       'Accounts.AccountRefMonth', 'Accounts.NextDueDate',
       'Accounts.LastMadeUpDate', 'Accounts.AccountCategory',
       'Returns.NextDueDate', 'Returns.LastMadeUpDate',
       'Mortgages.NumMortCharges', 'Mortgages.NumMortOutstanding',
       'Mortgages.NumMortPartSatisfied', 'Mortgages.NumMortSatisfied',
       'SICCode.SicText_1', 'SICCode.SicText_2', 'SICCode.SicText_3',
       'SICCode.SicText_4', 'LimitedPartnerships.NumGenPartners',
       'LimitedPartnerships.NumLimPartners', 'URI', 'PreviousName_1.CONDATE',
       ' PreviousName_1.CompanyName', ' PreviousName_2.CONDATE',
       ' PreviousName_2.C

<IPython.core.display.Javascript object>

In [6]:
df.shape

(5080061, 55)

<IPython.core.display.Javascript object>

In [7]:
df["CompanyStatus"].value_counts() * 100 / df.shape[0]

Active                                              93.031383
Active - Proposal to Strike off                      4.879252
Liquidation                                          1.986984
In Administration                                    0.048070
Live but Receiver Manager on at least one charge     0.025551
Voluntary Arrangement                                0.013583
In Administration/Administrative Receiver            0.005728
RECEIVERSHIP                                         0.003740
ADMINISTRATION ORDER                                 0.002539
ADMINISTRATIVE RECEIVER                              0.002165
In Administration/Receiver Manager                   0.000787
RECEIVER MANAGER / ADMINISTRATIVE RECEIVER           0.000197
VOLUNTARY ARRANGEMENT / ADMINISTRATIVE RECEIVER      0.000020
Name: CompanyStatus, dtype: float64

<IPython.core.display.Javascript object>

In [8]:
df["Sector1"] = df["SICCode.SicText_1"].str[:2].map(CLASSIFICATIONS)
df["Sector2"] = df["SICCode.SicText_2"].str[:2].map(CLASSIFICATIONS)
df["Sector3"] = df["SICCode.SicText_3"].str[:2].map(CLASSIFICATIONS)
df["Sector4"] = df["SICCode.SicText_4"].str[:2].map(CLASSIFICATIONS)

<IPython.core.display.Javascript object>

In [9]:
df["Sector1"].value_counts() * 100 / df.shape[0]

Professional, Scientific and Technical Activities                       13.163070
Wholesale and Retail Trade; Repair of Motor Vehicles and Motorcycles    12.442252
Construction                                                            10.259660
Real Estate Activities                                                   8.458461
Information and Communication                                            8.370766
Administrative and Support Service Activities                            8.276082
Accommodation and Food Service Activities                                4.847048
Other Service Activities                                                 4.617071
Manufacturing                                                            4.498588
Human Health and Social Work Activities                                  3.787238
Financial and Insurance Activities                                       3.482261
Transportation and Storage                                               3.322165
Activities of Ex

<IPython.core.display.Javascript object>

In [10]:
df[df["Sector1"] == "Manufacturing"].shape

(228531, 59)

<IPython.core.display.Javascript object>

* Manufacturing companies from SICCode_1 = 228,531

In [11]:
df[
    (df["Sector1"] == "Manufacturing")
    | (df["Sector2"] == "Manufacturing")
    | (df["Sector3"] == "Manufacturing")
    | (df["Sector4"] == "Manufacturing")
].shape

(230231, 59)

<IPython.core.display.Javascript object>

* Manufacturing companies from SICCode_1,SICCode_2,SICCode_3,SICCode_4  = 230,231
* 1,700 more, compared to previous case.

In [12]:
df_m = df[
    (df["Sector1"] == "Manufacturing")
    | (df["Sector2"] == "Manufacturing")
    | (df["Sector3"] == "Manufacturing")
    | (df["Sector4"] == "Manufacturing")
]
df_m["CompanyStatus"].value_counts() * 100 / df_m.shape[0]

Active                                              92.237796
Active - Proposal to Strike off                      4.842093
Liquidation                                          2.701634
In Administration                                    0.133344
Voluntary Arrangement                                0.032142
In Administration/Administrative Receiver            0.020849
RECEIVERSHIP                                         0.014333
ADMINISTRATIVE RECEIVER                              0.009556
ADMINISTRATION ORDER                                 0.005212
Live but Receiver Manager on at least one charge     0.002606
In Administration/Receiver Manager                   0.000434
Name: CompanyStatus, dtype: float64

<IPython.core.display.Javascript object>

In [13]:
df_ac = df_m[df_m["CompanyStatus"] == "Active"]
df_ac.shape

(212360, 59)

<IPython.core.display.Javascript object>

* 212,360 Active Companies

In [14]:
fin = pd.read_csv("financial_data1.csv", low_memory=False)
fin.shape

(25419309, 14)

<IPython.core.display.Javascript object>

In [15]:
df_ac[" CompanyNumber"] = df_ac[" CompanyNumber"].apply(lambda x: x.zfill(8))
fin["Company Registered Number"] = fin["Company Registered Number"].apply(
    lambda x: x.zfill(8)
)

C:\Users\VRUser2\AppData\Local\Temp\ipykernel_14716\3423395216.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ac[' CompanyNumber'] = df_ac[' CompanyNumber'].apply(lambda x: x.zfill(8))


<IPython.core.display.Javascript object>

In [16]:
df_ac[df_ac[" CompanyNumber"].isin(fin["Company Registered Number"])].shape

(150732, 59)

<IPython.core.display.Javascript object>

In [18]:
150732 / df_ac.shape[0]

0.70979468826521

<IPython.core.display.Javascript object>

In [19]:
fin[fin["Company Registered Number"].isin(df_ac[" CompanyNumber"])].to_csv(
    "manf_finan.csv", header=True, index=False
)

<IPython.core.display.Javascript object>

In [22]:
manf_fin = pd.read_csv("manf_finan-cleaned-classified.csv")
manf_fin.shape

C:\Users\VRUser2\AppData\Local\Temp\ipykernel_14716\110682329.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  manf_fin = pd.read_csv("manf_finan-cleaned-classified.csv")


(147431, 7)

<IPython.core.display.Javascript object>

In [23]:
manf_fin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 147431 entries, 0 to 147430
Data columns (total 7 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   Company Registered Number  147431 non-null  object 
 1   Equity                     147431 non-null  float64
 2   Percentage Change          147431 non-null  object 
 3   SME Classification         147431 non-null  object 
 4   Equity Category            147431 non-null  object 
 5   PCT Category               147431 non-null  object 
 6   Category                   147431 non-null  object 
dtypes: float64(1), object(6)
memory usage: 7.9+ MB


<IPython.core.display.Javascript object>

In [25]:
manf_fin["Category"].value_counts()

Strugglers           68820
Prospects            38704
No Classification    33072
Contenders            5693
Champions              954
Star Performers        188
Name: Category, dtype: int64

<IPython.core.display.Javascript object>

In [26]:
manf_fin["Company Registered Number"] = manf_fin["Company Registered Number"].apply(
    lambda x: str(x).zfill(8)
)

<IPython.core.display.Javascript object>

In [27]:
df_ac[df_ac[" CompanyNumber"].isin(manf_fin["Company Registered Number"])].shape

(147431, 59)

<IPython.core.display.Javascript object>

In [28]:
pd.merge(
    df_ac,
    manf_fin,
    how="left",
    left_on=" CompanyNumber",
    right_on="Company Registered Number",
).info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 212360 entries, 0 to 212359
Data columns (total 66 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   CompanyName                         212360 non-null  object 
 1    CompanyNumber                      212360 non-null  object 
 2   RegAddress.CareOf                   2204 non-null    object 
 3   RegAddress.POBox                    586 non-null     object 
 4   RegAddress.AddressLine1             212358 non-null  object 
 5    RegAddress.AddressLine2            148626 non-null  object 
 6   RegAddress.PostTown                 210686 non-null  object 
 7   RegAddress.County                   87652 non-null   object 
 8   RegAddress.Country                  145157 non-null  object 
 9   RegAddress.PostCode                 212186 non-null  object 
 10  CompanyCategory                     212360 non-null  object 
 11  CompanyStatus             

<IPython.core.display.Javascript object>

In [29]:
manf_fin["Company Registered Number"].value_counts()

00019071    1
10911432    1
10910973    1
10911187    1
10911238    1
           ..
06769897    1
06769649    1
06769605    1
06769218    1
SC700594    1
Name: Company Registered Number, Length: 147431, dtype: int64

<IPython.core.display.Javascript object>